### PR TITLE
Fix data races

### PR DESCRIPTION
**Pull request by tribal-tec**

### CODE REVIEW

File: rockets/ws/messageHandler.h [hunk 1]
```diff
@@ -22,16 +22,24 @@
...
 {
+using Connections = std::map<lws*, ConnectionPtr>;
+
 /**
  * Handle message callbacks for text/binary messages.
  */
 class MessageHandler
 {
 public:
+    MessageHandler() = default;
```
[the remark this answers is not in the document]
Comment (tribal-tec) [reply rockets/ws/messageHandler.h]: The ws client also uses the message handler, so yes. Or the empty connections are moved to the client, which is equally strange.

File: rockets/ws/messageHandler.cpp [hunk 2]
```diff
@@ -99,48 +99,44 @@ void MessageHandler::handleCloseConnection(ConnectionPtr connection)
...
     {
-        connections = _connections;
-        auto end = std::remove_if(connections.begin(), connections.end(),
-                                  [&sender](auto conn) {
-                                      return conn.lock() == sender;
-                                  });
-        connections.erase(end, connections.end());
+        for (const auto& connection : _connections)
+        {
+            if (response.recipient == Recipient::others &&
```
[the remark this answers is not in the document]
Comment (tribal-tec) [reply rockets/ws/messageHandler.cpp]: The 'case' handles both, `all` and `others`, so I think I have to keep the check for `others`?!

[the remark this answers is not in the document]
Comment (rdumusc) [reply rockets/ws/messageHandler.cpp]: ah yes, I didn't read correctly ;-)

File: rockets/ws/messageHandler.h (hunk 1 above)
[the remark this answers is not in the document]
Comment (rdumusc) [reply rockets/ws/messageHandler.h]: Hmm I see, looks like this implies deeper refactorings. What does it mean for the client to send a response to "all" or "others" when it's connected to just one server anyway? Another time, then.

**Pull request merged**